packages/neon/neon/lib/src/pages/login_check_server_status.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:neon/src/blocs/login_check_server_status.dart';
+import 'package:neon/src/widgets/exception.dart';
+import 'package:neon/src/widgets/linear_progress_indicator.dart';
+import 'package:neon/src/widgets/validation_tile.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:provider/provider.dart';
+
+class LoginCheckServerStatusPage extends StatefulWidget {
+  const LoginCheckServerStatusPage({
+    required this.serverURL,
+    super.key,
+  });
+
+  final String serverURL;
+
+  @override
+  State<LoginCheckServerStatusPage> createState() => _LoginCheckServerStatusPageState();
+}
+
+class _LoginCheckServerStatusPageState extends State<LoginCheckServerStatusPage> {
+  late final LoginCheckServerStatusBloc bloc;
+
+  @override
+  void initState() {
+    super.initState();
+
+    bloc = LoginCheckServerStatusBloc(Provider.of<PackageInfo>(context, listen: false), widget.serverURL);
+  }
+
+  @override
+  Widget build(final BuildContext context) => Scaffold(
+        appBar: AppBar(),
+        body: Center(
+          child: Padding(
+            padding: const EdgeInsets.all(10),
+            child: ResultBuilder.behaviorSubject(
+              stream: bloc.state,
+              builder: (final context, final state) => Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  NeonLinearProgressIndicator(
+                    visible: state.isLoading,
+                  ),
+                  NeonException(
+                    state.error,
+                    onRetry: bloc.refresh,
+                  ),
+                  _buildServerVersionTile(state),
+                  _buildMaintenanceModeTile(state),
+                  Align(
+                    alignment: Alignment.bottomRight,
+                    child: ElevatedButton(
+                      onPressed: state.hasData && state.requireData.isSupported && !state.requireData.maintenance
+                          ? () => Navigator.of(context).pop(true)
+                          : null,
+                      child: Text(AppLocalizations.of(context).actionContinue),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+
+  Widget _buildServerVersionTile(final Result<CoreServerStatus> result) {
+    final (state, title) = !result.hasData
+        ? (ValidationState.loading, AppLocalizations.of(context).loginCheckingServerVersion)
+        : result.requireData.isSupported
+            ? (
+                ValidationState.success,
+                AppLocalizations.of(context).loginSupportedServerVersion(result.requireData.versionstring),
+              )
+            : (
+                ValidationState.failure,
+                AppLocalizations.of(context).loginUnsupportedServerVersion(result.requireData.versionstring),
+              );
```

Comment:
This looks a bit funny :)

It's probably worth directly returning the ValidationTiles in normal if clauses (I hate the readability of nested `?` conditions).

Reply:
Oh yeah, that makes so much more sense!